fastlane/lib/fastlane/plugins/plugin_generator.rb

```diff
@@ -0,0 +1,109 @@
+require 'fileutils'
+require 'erb'
+
+module Fastlane
+  class PluginGenerator
+    def initialize(ui             = PluginGeneratorUI.new,
+                   info_collector = PluginInfoCollector.new(ui))
+      @ui = ui
+      @info_collector = info_collector
+    end
+
+    # entry point
+    def generate(plugin_name = nil)
+      plugin_info = @info_collector.collect_info(plugin_name)
+
+      generate_paths(plugin_info)
+
+      generate_dot_rspec(plugin_info)
+      generate_dot_gitignore(plugin_info)
+      generate_gemfile(plugin_info)
+      generate_gemspec(plugin_info)
+      generate_plugin_rb(plugin_info)
+      generate_readme(plugin_info)
+      generate_version(plugin_info)
+      generate_license(plugin_info)
+      generate_action(plugin_info)
+      generate_helper(plugin_info)
+      generate_spec_helper(plugin_info)
+      generate_action_spec(plugin_info)
+      generate_rakefile(plugin_info)
+
+      @ui.success "\nYour plugin was successfully generated at #{plugin_info.gem_name}/ 🚀"
+      @ui.success "\nTo get started with using this plugin, run"
+      @ui.message "\n    fastlane add_plugin #{plugin_info.plugin_name}\n"
+      @ui.success "\nfrom a fastlane-enabled app project directory and provide the following as the path:"
+      @ui.message "\n    #{File.expand_path(plugin_info.gem_name)}\n\n"
```

Comment:
We had spaced it out because we thought it highlighted the info we want you to see (the command and the path) a bit better. We can squash it if you like that better 👍